Sources/TSCUtility/SerializedDiagnostics.swift

```diff
@@ -185,7 +185,7 @@ extension SerializedDiagnostics {
 
   public struct SourceLocation: Equatable {
     /// The filename associated with the diagnostic.
-    public var filename: String
+    public var filename: String?
```

Comment:
What do the line and column mean (semantically) in the absence of a filename?  If the .dia still has those fields without the filename then this does seem like the right change, but what should an IDE (for example) do with such a source location?

Reply:
Or should this be modeled so that it's the `SourceLocation` itself that is optional?  (even if that's indicated by just a `nil` filename field in the file).

Reply:
I updated this to just make the entire SourceLocation optional. Currently, Swift uses the absence of a filename to indicate either a missing or invalid location, and it's always paired with a line/column of 0. I'm fairly certain diagnostics formatted from Clang will look the same, but even if it encodes additional info there's really nothing an IDE can do to present that in a useful way.

Reply:
That makes a lot of sense to me.  Thanks!